### PR TITLE
Add AID Activation by Interval-wise Dropout

### PR DESCRIPTION
**Pull request by rockerBOO**
Activation by Interval-wise Dropout: A Simple Way to Prevent Neural Networks from Plasticity Loss
https://arxiv.org/abs/2502.01342

> Plasticity loss, a critical challenge in neural net-
work training, limits a model’s ability to adapt to
new tasks or shifts in data distribution. This pa-
per introduces AID (Activation by Interval-wise
Dropout), a novel method inspired by Dropout, de-
signed to address plasticity loss. Unlike Dropout,
AID generates subnetworks by applying Dropout
with different probabilities on each preactivation
interval. Theoretical analysis reveals that AID reg-
ularizes the network, promoting behavior analo-
gous to that of deep linear networks, which do not
suffer from plasticity loss.

| Architecture |  |
|--------|--------|
| Screenshot 2025-04-13 at 14-00-29 Activation by Interval-wise Dropout A Simple Way to Prevent Neural Networks from Plasticity Loss - 2502 01342v1 pdf | Screenshot 2025-04-13 at 14-00-08 Activation by Interval-wise Dropout A Simple Way to Prevent Neural Networks from Plasticity Loss - 2502 01342v1 pdf | 

## Comparing to other regularization techniques
Screenshot 2025-04-13 at 14-03-36 Activation by Interval-wise Dropout A Simple Way to Prevent Neural Networks from Plasticity Loss - 2502 01342v1 pdf

*Note:* Only currently implemented on lora_flux.

```
network_module = "networks.lora_flux"
network_args = [
    "aid_dropout=0.9"
]
```

```
--network_module networks.lora_flux --network_args aid_dropout=0.9
```

### CODE REVIEW

**Comment by rockerBOO** [comment]
AID requires inference changes as well as training. I have modifications for comfy to support it but will try to upstream those as well as share them here. Otherwise it diverges a lot more if it is not applied at inference. 

| AID | Without AID | Base |
|--------|--------|--------|
| c-f1-2025-04-17_00114_ |  c-f1-2025-04-17_00066_ | c-f1-2025-04-17_00065_ |
| c-f1-2025-04-17_00108_ | c-f1-2025-04-17_00045_  | c-f1-2025-04-17_00046_ |
| c-f1-2025-04-17_00123_ | c-f1-2025-04-17_00027_ | c-f1-2025-04-17_00025_ | 
|  c-f1-2025-04-17_00126_ | c-f1-2025-04-17_00014_ | c-f1-2025-04-17_00013_

**Comment by rockerBOO** [comment]
@67372a  https://gist.github.com/rockerBOO/f7325ddb2d33c05cbd1b672b7ba62917 Here is the patch for comfy. It's probably unstable if they change any of the implementation in here but should work for the meantime.

**Comment by rockerBOO** [comment]
Of course they update the LoRA implementation just now :).

Updated patch (ignore/rollback the previous patch): 
https://gist.github.com/rockerBOO/a03d169443db24a37039c83ae1c58253

**Comment by sangyeon-park** [comment]
Hello, I am the author of the paper (AID).  I was both surprised and delighted to see that my methodology was used. I have recently submitted a revised version of the paper, which includes considerations relevant to its implementation. Please feel free to reach out if you need it. If you have any additional questions, I would be happy to discuss any further matters.
